numba/tests/test_boundscheck.py

```diff
@@ -120,6 +120,10 @@ def test_no_cuda_boundscheck(self):
             def func():
                 pass
 
+        @cuda.jit(boundscheck=False)
```

Comment:
Thanks for adding this, perhaps also add a comment as to why this is now present as I'm pretty sure we won't remember in a few months time :) Thanks!